alfresco-search/src/test/java/org/alfresco/repo/search/impl/lucene/analysis/MultiTermSynonymsTest.java

```diff
@@ -0,0 +1,278 @@
+                    synonymsFactory.inform(resourceLoader);
+                    TokenStream synonymsFilter = synonymsFactory.create(tokenizer);
+
+                    Map<String, String> stopFilterConfig = new HashMap<>();
+                    stopFilterConfig.put("words", "stopwords.txt");
+                    stopFilterConfig.put("ignoreCase", "true");
+
+                    SynonymAwareStopFilterFactory stopWordsFactory = new SynonymAwareStopFilterFactory(stopFilterConfig);
+                    stopWordsFactory.inform(resourceLoader);
+                    TokenStream sinkStream = stopWordsFactory.create(synonymsFilter);
+
+                    return new TokenStreamComponents(tokenizer, sinkStream);
+                }
+                catch (Exception exception)
+                {
+                    throw new IllegalArgumentException(exception);
+                }
+            }
+        };
+    }
+
+    /**
+     * No synonyms and no stopwords (plain and subsequent tokens with no removal).
+     * No Graph is being generated from the outcoming stream.
+     */
+    @Test
+    public void noSynonmsNoStopwords() throws Exception
+    {
+        final String text = "no synonyms and no stopwords";
+        final List<PackedTokenAttributeImpl> expected = asList(
+                token("no", 0, 2, 1, 1, ALPHANUM_TYPE),
+                token("synonyms", 3, 11, 1, 1, ALPHANUM_TYPE),
+                token("and", 12, 15, 1, 1, ALPHANUM_TYPE),
+                token("no", 16, 18, 1, 1, ALPHANUM_TYPE),
+                token("stopwords", 19, 28, 1, 1, ALPHANUM_TYPE));
+
+        assertAnalysisCorrectness(text, expected);
+    }
+
+    /**
+     * A synonym is detected and the correct graph is generated.
+     * Note the original text (out *of* warranty) contains a stopwords but being that part of a synonym, it won't be
+     * removed so a the graph will later generate a correct phrase query (i.e. "out of warranty")
+     */
+    @Test
+    public void multiTermConceptWithSynonyms() throws Exception
+    {
+        final String text = "Car is Out of warranty";
+        final List<PackedTokenAttributeImpl> expected = asList(
+                token("Car", 0, 3, 1, 1, ALPHANUM_TYPE),
+                token("is", 4, 6, 1, 1, ALPHANUM_TYPE),
+                token("oow", 7, 22, 1, 3, SYNONYM_TYPE),
+                token("Out", 7, 10, 0, 1, ALPHANUM_TYPE),
+                token("of", 11, 13, 1, 1, ALPHANUM_TYPE),
+                token("warranty", 14, 22, 1, 1, ALPHANUM_TYPE));
+
+        assertAnalysisCorrectness(text, expected);
+    }
+
+    /**
+     * This is the same test as above but instead of starting from the expanded form, the original text contains the
+     * acronym (oow). Expected result is the same, the only difference is in the different tokens types.
+     */
+    @Test
+    public void acronymsIsExpandedAndContainsStopwords() throws Exception
```

Comment:
These tests are great. The only thing I can think of that could be added is that I don't think there's a test for something that _does_ contain a stopword.

Reply:
I'm not sure I got you, the previous test ("_Car is Out of warranty_") isn't what you're looking for?
Could you please write a text example? I'd happy to add more cases

Reply:
Actually I saw that there are some unit tests like this in the other new test file.